src/utils/wallets/wallets.provider.tsx

```diff
@@ -65,17 +65,17 @@ export function WalletsProvider({
 
       let walletStatus: WalletStatus = "noWallets";
 
-      if (hasWallets && decryptionKey) {
-        walletStatus = "unlocked";
-      } else if (!decryptionKey) {
-        walletStatus = "locked";
+      if (hasWallets) {
+        walletStatus = decryptionKey ? "unlocked" : "locked";
       } else if (redirectToWelcome) {
         // This should only happen when opening the regular popup, but not for the auth popup, as the
         // `createAuthPopup` will open the welcome page directly, instead of the popup, if needed:
 
         openOrSelectWelcomePage(true);
```

Comment:
@nicholaswma Any reason why an `await` here is necessary? When adding it, I see the extension popup appear for a fraction of a second, before the Welcome page opens on a new tab; without it, clicking the extension icon opens the Welcome page straight away without the popup flicker.